### ISSUE TITLE
Directory dependencies should not return UNC paths on Windows

### ISSUE THREAD

**Issue by asodeur** [sdispater/poetry#959]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Windows 7
- **Poetry version**: 0.12.11

Assume we have a network drive `\\mycompany.com\drive_z` mapped onto `Z:` on Windows. If you specify a directory dependency anywhere on the `Z:`-drive, `DirectoryDependency.full_path` will be the UNC path, ie `\\mycompany.com\drive_z\the_dependency`. (This is a `pathlib` 'feature', [see](https://bugs.python.org/issue32442)).

This will cause trouble the as soon as  `Env.run` is called on an `Env` for the path because `cmd.exe` does not support UNC working directories (you must use the mapped drive letters, ie `Z:\the_dependency`).

**Comment by dimaqq** [comment]
Perhaps network or cross-device paths can be refused altogether?

**Comment by asodeur** [comment]
That would be a serious limitation IMHO. Another way of coping with `pathlib`'s preference for UNC paths would be to ask Windows users to enable UNC paths in the shell which can be done in recent Windows versions via a registry setting:
```REG ADD "HKCU\Software\Microsoft\Command Processor" /v DisableUNCCheck /t REG_DWORD /d 1 /f```
But I am not sure this can be done without admin rights. Hence, I'd prefer the solution from the PR above.

**Comment by oramirite** [comment]
Hello there! I just got slightly bit by this. I moved a repository I've been working on over to a network drive and it seems Poetry doesn't support this based on encountering the same issue. I agree with @asodeur that this is a serious limitation.

Working around the issue by enabling UNC paths using that registry entry does work, demonstrating that there are no technical issues at play preventing this. It would be grand if Poetry simply resolved the mapped drive path correctly.
